docs/contributing/coding-guidelines/ros-nodes/parameters.md

```diff
@@ -1,36 +1,175 @@
+Find more information on parameters from the official ROS documentation:
 
-## Parameter files
+- [Understanding ROS 2 Parameters](https://docs.ros.org/en/humble/Tutorials/Beginner-CLI-Tools/Understanding-ROS2-Parameters/Understanding-ROS2-Parameters.html)
+- [About ROS 2 Parameters](https://docs.ros.org/en/humble/Concepts/About-ROS-2-Parameters.html)
 
-Autoware has the following two types of parameter files for ROS packages:
+## Workflow
 
-- **Node parameter file**
-  - Node parameter files store the default parameters provided for each package in Autoware.
-    - For example, [the parameter of `behavior_path_planner`](https://github.com/autowarefoundation/autoware.universe/tree/245242cee866de2d113e89c562353c5fc17f1f98/planning/behavior_path_planner/config)
-  - All nodes in Autoware must have a parameter file if one or more parameters that can be customized by the user are defined.
-  - For `FOO_package`, the parameter is expected to be stored in `FOO_package/config`.
-  - The launch file for individual packages must load node parameter by default:
+A ROS package which uses the [declare_parameter(...)](https://docs.ros.org/en/ros2_packages/humble/api/rclcpp/generated/classrclcpp_1_1Node.html#_CPPv4N6rclcpp4Node17declare_parameterERKNSt6stringERKN6rclcpp14ParameterValueERKN14rcl_interfaces3msg19ParameterDescriptorEb) function should:
 
-```xml
-<launch>
-  <arg name="foo_node_param_path" default="$(find-pkg-share FOO_package)/config/foo_node.param.yaml" />
+- use the [declare_parameter(...)](https://docs.ros.org/en/ros2_packages/humble/api/rclcpp/generated/classrclcpp_1_1Node.html#_CPPv4N6rclcpp4Node17declare_parameterERKNSt6stringERKN6rclcpp14ParameterValueERKN14rcl_interfaces3msg19ParameterDescriptorEb) with out a default value
+- create a parameter file
+- create a schema file
 
-  <node pkg="FOO_package" exec="foo_node">
-    ...
-    <param from="$(var foo_node_param_path)" />
-  </node>
-</launch>
+The rationale behind this workflow is to have a verified single source of truth to pass to the ROS node and to be used in the web documentation. The approach reduces the risk of using invalid parameter values and makes maintenance of documentation easier. This is achieved by:
+
+- [declare_parameter(...)](https://docs.ros.org/en/ros2_packages/humble/api/rclcpp/generated/classrclcpp_1_1Node.html#_CPPv4N6rclcpp4Node17declare_parameterERKNSt6stringERKN6rclcpp14ParameterValueERKN14rcl_interfaces3msg19ParameterDescriptorEb) throws an exception if an expected parameter is missing in the parameter file
+- the schema validates the parameter file in the CI and renders a parameter table, as depicted in the graphics below
+
+  ```mermaid
+  flowchart TD
+      NodeSchema[Schema file: *.schema.json]
+      ParameterFile[Parameter file: *.param.yaml]
+      WebDocumentation[Web documentation table]
+
+      NodeSchema -->|Validation| ParameterFile
+      NodeSchema -->|Generate| WebDocumentation
+  ```
+
+*
```

Comment:
The pre-commit ci is complaining about this line, i think it's a typo.

Reply:
Yes, it is, removing it!